pkg/cephmgr/mds/agent.go

```diff
@@ -119,38 +118,39 @@ func (a *mdsAgent) DestroyLocalService(context *clusterd.Context) error {
 	return nil
 }
 
-// create a keyring for the mds client with a limited set of privileges
-func createMDSKeyring(conn client.Connection, id, configDir string) error {
+func getKeyringProperties(id string) (string, []string) {
 	username := fmt.Sprintf("mds.%s", id)
-	keyringPath := getMDSKeyringPath(configDir, id)
 	access := []string{"osd", "allow *", "mds", "allow", "mon", "allow profile mds"}
+	return username, access
+}
+
+// create a keyring for the mds client with a limited set of privileges
+func createKeyringAndSave(conn client.Connection, id, configDir string) error {
+	username, access := getKeyringProperties(id)
+	keyringPath := getMDSKeyringPath(configDir, id)
 	keyringEval := func(key string) string {
 		return fmt.Sprintf(keyringTemplate, id, key)
 	}
 
 	return mon.CreateKeyring(conn, username, keyringPath, access, keyringEval)
 }
 
+// create a keyring for the mds client with a limited set of privileges
+func CreateKeyring(conn client.Connection, id string) (string, error) {
+	// get-or-create-key for the user account
+	username, access := getKeyringProperties(id)
+	keyring, err := client.AuthGetOrCreateKey(conn, username, access)
+	if err != nil {
+		return "", fmt.Errorf("failed to get or create auth key for %s. %+v", username, err)
+	}
+
+	return keyring, nil
+}
+
 func (a *mdsAgent) startMDS(context *clusterd.Context, cluster *mon.ClusterInfo, id string) error {
 
-	// start the monitor daemon in the foreground with the given config
-	logger.Infof("starting mds %s", id)
-
-	confFile := getMDSConfFilePath(context.ConfigDir, id, cluster.Name)
-	util.WriteFileToLog(logger, confFile)
-
-	mdsNameArg := fmt.Sprintf("--name=mds.%s", id)
-	mdsProc, err := context.ProcMan.Start(
-		fmt.Sprintf("mds%s", id),
-		"mds",
-		regexp.QuoteMeta(mdsNameArg),
-		proc.ReuseExisting,
-		"--foreground",
-		mdsNameArg,
-		fmt.Sprintf("--cluster=%s", cluster.Name),
-		fmt.Sprintf("--conf=%s", confFile),
-		fmt.Sprintf("--keyring=%s", getMDSKeyringPath(context.ConfigDir, id)),
-		"-i", id)
+	config := &Config{ID: id, InProc: false, ClusterInfo: cluster}
```

Comment:
When running as a k8s daemon, we want to run the mds daemon in proc. When we're running on the rook orchestrator, we want to start a child process